lib/ansible/modules/cloud/vmware/vmware_guest_snapshot.py

```diff
@@ -60,6 +60,7 @@
    folder:
         description:
             - Define instance folder location.
+        required: True
```

Comment:
optional parameters should always have a default, the module does not need to use it though.

A saner default would have been 'null/None' in which case the module should ignore it.